packages/web-react/src/components/Grid/__tests__/Grid.test.tsx

```diff
@@ -14,39 +14,114 @@ describe('Grid', () => {
+    const testId = 'grid-test-id';
+
+    render(<Grid cols={{ mobile: 2, tablet: 3, desktop: 4 }} data-testid={testId} />);
+
+    const element = screen.getByTestId(testId) as HTMLElement;
 
-    const element = dom.container.querySelector('div') as HTMLElement;
     expect(element).toHaveClass('Grid--cols-2');
     expect(element).toHaveClass('Grid--tablet--cols-3');
     expect(element).toHaveClass('Grid--desktop--cols-4');
   });
 
   it('should have cols classname', () => {
-    const dom = render(<Grid cols={2} />);
+    const testId = 'grid-test-id';
+
+    render(<Grid cols={2} data-testid={testId} />);
+
+    const element = screen.getByTestId(testId) as HTMLElement;
 
-    const element = dom.container.querySelector('div') as HTMLElement;
     expect(element).toHaveClass('Grid--cols-2');
   });
+
+  it('should render with custom spacing', () => {
+    const testId = 'grid-test-id';
+
+    render(<Grid spacing="space-600" data-testid={testId} />);
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-600)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-600)' });
+  });
+
+  it('should render with custom spacing for each breakpoint', () => {
+    const testId = 'grid-test-id';
+
+    render(
+      <Grid spacing={{ mobile: 'space-100', tablet: 'space-1000', desktop: 'space-1200' }} data-testid={testId} />,
+    );
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-x-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-x-desktop': 'var(--spirit-space-1200)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-desktop': 'var(--spirit-space-1200)' });
+  });
+
+  it('should render with custom vertical spacing for each breakpoint', () => {
+    const testId = 'grid-test-id';
+
+    render(
+      <Grid spacingY={{ mobile: 'space-100', tablet: 'space-1000', desktop: 'space-1200' }} data-testid={testId} />,
+    );
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-desktop': 'var(--spirit-space-1200)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-100)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x-tablet': 'var(--spirit-space-1000)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x-desktop': 'var(--spirit-space-1200)' });
```

Comment:
Isn't it safer to check there is the default value (800) for the x axis…? 🤔

Reply:
I'm afraid that [`toHaveStyle()` method](https://github.com/testing-library/jest-dom#tohavestyle) checks only styles set via `style` attribute in the HTML. It doesn't evaluate applied CSS styles as well…